fe/fe-core/src/test/java/com/starrocks/persist/metablock/SRMetaBlockV1Test.java

```diff
@@ -366,4 +373,64 @@ public void emptyFile() throws Exception {
         }
         dis.close();
     }
+
+    public static class TestTable {
+        @SerializedName("clazz")
+        private String clazz = "unknownTable";
+
+        public TestTable() {
+        }
+    }
+
+    @Test
+    public void testSubTypeRollbackInReadCollection() throws Exception {
+        UtFrameUtils.PseudoImage image = new UtFrameUtils.PseudoImage();
+        SRMetaBlockWriter writer = new SRMetaBlockWriterV1(image.getDataOutputStream(),
+                SRMetaBlockID.LOCAL_META_STORE, 5);
+        writer.writeInt(4);
+        writer.writeJson(new MysqlTable());
+        writer.writeJson(new HiveTable());
+        writer.writeJson(new JDBCTable());
+        writer.writeJson(new TestTable());
+        writer.close();
+
+        Config.ignore_unknown_subtype = true;
+        List<Table> tables = new ArrayList<>();
+        SRMetaBlockReader reader = new SRMetaBlockReaderV1(image.getDataInputStream());
+        reader.readCollection(Table.class, tables::add);
+        reader.close();
+
+        Assert.assertEquals(3, tables.size());
+        Assert.assertTrue(tables.get(0) instanceof MysqlTable);
+        Assert.assertTrue(tables.get(1) instanceof HiveTable);
+        Assert.assertTrue(tables.get(2) instanceof JDBCTable);
+    }
+
+    @Test
+    public void testSubTypeRollbackInMapEntry() throws Exception {
+        UtFrameUtils.PseudoImage image = new UtFrameUtils.PseudoImage();
+        SRMetaBlockWriter writer = new SRMetaBlockWriterV1(image.getDataOutputStream(),
+                SRMetaBlockID.LOCAL_META_STORE, 9);
+        writer.writeInt(4);
+        writer.writeLong(1);
+        writer.writeJson(new MysqlTable());
+        writer.writeLong(2);
+        writer.writeJson(new HiveTable());
+        writer.writeLong(3);
+        writer.writeJson(new JDBCTable());
+        writer.writeLong(4);
+        writer.writeJson(new TestTable());
+        writer.close();
+
+        Config.ignore_unknown_subtype = true;
+        Map<Long, Table> tables = new HashMap<>();
+        SRMetaBlockReader reader = new SRMetaBlockReaderV1(image.getDataInputStream());
```

Comment:
1. maybe add test to cover the case that `Config.ignore_unknown_subtype` is set to false, The UnknownSubtype Exception is expected to throw.

2. need to reset the `Config.ignore_unknown_subtype` to default value either after case is done or in `tearDown`